File: Ruby_api_samples/create_solution_category.rb
#create solution category

require "rubygems"
require "rest_client"
require "json"

site = RestClient::Resource.new("http://domain.freshdesk.com/solution/categories.json","sample@freshdesk.com","test")

response = site.post({:solution_category=>{:name=>"new category",:description=>"Testing solution category"}},:content_type=>"application/json")

puts "response: #{response.code} \n #{response.body} \n"

File: google_script/create_solution_category.js
function testCreateCategory() {

  var headers = {
    'Content-type': 'application/json',
    'Authorization': 'Basic TOKEN:X_BASE64ENCODED='
  };
  
  var payload = '{"solution_category":{"name":"API test","description":"API related documents"},"tags":{"name":""}}';
  
  var url = 'http://DOMAIN.freshdesk.com/solution/categories.json';
  var options = {
    'method': 'post',
    'headers': headers,
    'payload': payload,
    muteHttpExceptions: true
  };
  var response = UrlFetchApp.fetch(url, options);
  
  Logger.log("code: " + response.getResponseCode());
  Logger.log("text: " + response.getContentText());

}
